langchain/callbacks/promptlayer_callback.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+from typing import Any, Dict, List, Optional
+
+from promptlayer.utils import get_api_key, promptlayer_api_request
+
+from langchain.callbacks.base import BaseCallbackHandler
+from langchain.chat_models import ChatOpenAI
+from langchain.llms import OpenAI, Replicate
+from uuid import UUID
+from langchain.schema import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+    LLMResult,
+    SystemMessage,
+)
+
+
+class PromptLayerHandler(BaseCallbackHandler):
+    def _convert_message_to_dict(self, message):
+        if isinstance(message, HumanMessage):
+            message_dict = {"role": "user", "content": message.content}
+        elif isinstance(message, AIMessage):
+            message_dict = {"role": "assistant", "content": message.content}
+        elif isinstance(message, SystemMessage):
+            message_dict = {"role": "system", "content": message.content}
+        else:
```

Comment:
You'll want to handle `ChatMessage` as well

Reply:
@vowelparrot what is the role for a `ChatMessage` I am having difficulty finding it in the docs

Reply:
It's a message with an arbitrary speaker role